### ISSUE TITLE
Is the project alive? What plans does the core team regarding the sophisticated terminal emulator?

### ISSUE THREAD

**Issue by xikxp1** [nushell/nana#77]
Thank you for developer such an extraordinary project like nushell.

The main benefit of nushell for me is its structured output and terminal emulator which extends this idea further is a perfect synergy for it.

Are there any official plans on nana development?

Thank you in advance!

**Comment by fdncred** [comment]
I'm not sure I'd call it dead but `nana` was an experiment to build nushell into a gui. It was a success but I'm not sure what else we can do at the moment since it appears that we really don't have anyone who wants to contribute to it.

We still want a nushell notebook type tool, minus jupyter. Nana was kind of a step in that direction.

For me personally, I'm still waiting for rust gui tools to mature. I've looked at dozens and don't really like any of them for building beautiful guis.

**Comment by darknight** [comment]
Just curious, have you ever gave a try on [slint](https://sling.dev)?

**Comment by fdncred** [comment]
@darknight Yes, I've tried slint but I believe it has an incompatible license.

**Comment by alicealysia** [comment]
@fdncred Hello, it is I, future Alysia! Here from the magical land of 2026

good news, it is not an incompatible license! The licenses folder lists compatible licenses, and among them is the MIT license.

I also think it's worth noting that tesseria has to my mind... the right structure.

Like, generally, when I think of easy to use gui frameworks, I think of react. React's main drawcard being its ability to manage state, events and views all from the same heirachy. This is something that tesseria does uniquely well in my opinion.

I also think relm4 is better than many think, however it's often overlooked because it uses macros to embed the gtk blueprint language and doesn't do the best job of explaining that.

**Comment by fdncred** [comment]
I'd have to see some evidence of that. I look at their page and see pricing and gplv3 but maybe it's hidden away somewhere else.

**Comment by alicealysia** [comment]
Okay, took a closer look at it, and yes, the option for licensing slint itself as open source is GPLv3

That said, GPLv3 is cross compatible with the MIT license.

They helpfully cover off how this specific licensing interaction works on their licensing FAQ:

https://github.com/slint-ui/slint/blob/master/FAQ.md#scenario-alice-is-a-software-developer-she-wants-her-code-to-be-licensed-under-mit-she-is-developing-an-application-aliceapp-that-links-to-slint-gplv3-alice-also-wants-to-allow-that-bob-a-user-of-aliceapp-can-fork-aliceapp-into-a-proprietary-application-called-bobapp

Bobapp... bobcat? Idk.

Anyways, yeah, you can use it, the trick here would be to ensure that you don't *directly* redistribute the source code in your app (easy enough, just include it as a rust module so that it can be installed directly when a user wants to compile it)

While it states that the "overall project must be GPLv3"

That's only if you want to redistribute the source code of slint itself.

What you're developing would be considered a module due to how cargo packages things.

Just. Ya know. Don't include a copy of slint and you should be fine. (Huh... so that's why we invented package managers lol(

**Comment by Diogenesoftoronto** [comment]
I think there are much better option in the ecosystem than slint now thankfully. Including the project from Zed, GPUI.

**Comment by fdncred** [comment]
I recently implemented this using the zed gpui https://github.com/fdncred/nu_plugin_to_gui

<img width="1155" height="621" alt="Image" src="https://github.com/user-attachments/assets/fec50c82-1afc-4e37-8f7a-bbd59711cd0a" />
